# Profitable App Profiles for the App Store and Google Play Markets


#### Objectives:

Early stage project , aimed to make us familiar with the Jupyter functions and work with pure Python functions to analyze data
The goal would be to determine a good market for a new app, and maybe determining what kind of app would be a good idea to produce.

##### Resources used:

**Anaconda distribution** - *Jupyter Notebook v 5.7.8*, *Python 3.7.3*


In [10]:
#reading files with csv reader

from csv import reader

file1 = open("AppleStore.csv",encoding ='utf8')
file2 = open("googleplaystore.csv", encoding = 'utf8')
apple = list(reader(file1))
google = list(reader(file2))
explore_data(apple,1,5)
explore_data(google,1,5)
explore_data(apple[1:],1,2,True)
explore_data(google[1:],1,2,True)
print(apple[0])
print(google[0])



['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', 

In [9]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [11]:
print(google[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [12]:
del google[10473]

Searching for duplicate values inside the google dataset


In [14]:
duplicates = []
no_duplicates = 0
for row in google:
    if row in duplicates:
        no_duplicates +=1
    else:
        duplicates.append(row)
print(no_duplicates)



483


So we have 483 duplicate entries in the google dataset, for some reason. We will write a custom function to clean the datasets.

In [17]:
reviews_max = {}
for row in google[1:]:    
    name = row[0]
    n_reviews = float(row[3])
    if name not in reviews_max:
        reviews_max[name]= n_reviews
    elif reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews

android_clean = []
already_added = []
for row in google[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(row)
        already_added.append(name)

explore_data(android_clean,1,5,True)
    

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


A quick function to determine if the name contains "ilegal" characters.

In [18]:
def try_(string):
    no = 0
    for letter in string:
        
        if ord(letter) > 127:
            no+=1
            if no >3:
                return False
    return True

try_('Instagram')


True

In [19]:
try_('爱奇艺PPS -《欢乐颂2》电视剧热播')


False

In [20]:
try_('Docs To Go™ Free Office Suite')


True

In [21]:
try_('Instachat 😜')

True

We will try now to isolate free apps in separate lists.

In [22]:
freeappsandroid = []
freeappsios= []
for row in android_clean:
    if row[6] == "Free" :
        freeappsandroid.append(row)
for row in apple[1:]:
    if float(row[4]) == 0:
        freeappsios.append(row)
print(len(freeappsandroid))
print(len(freeappsios))
    

8904
4056


Ok, apparenlty now we need to find a good genre for our new free app. Looking into apple and android most popular genres

In [25]:
def freq_table(dataset,index):
    freq_table = {}
    for row in dataset:
        if row[index] in freq_table:
            freq_table[row[index]] +=1
        else:
            freq_table[row[index]] =1
    for pair in freq_table:
        freq_table[pair] /= len(dataset)
        freq_table[pair] *=100
    return freq_table
    

In [26]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Top apps for Android:

In [27]:
display_table(android_clean,9)


Tools : 8.572315974738585
Entertainment : 5.808054664043897
Education : 5.280049694585361
Business : 4.348276219070297
Medical : 4.089450253649446
Personalization : 3.892742519929599
Productivity : 3.8720364426959315
Lifestyle : 3.8099182109949266
Finance : 3.571798322807744
Sports : 3.4268557821720678
Communication : 3.2612071643027227
Action : 3.095558546433378
Health & Fitness : 2.9816751216482036
Photography : 2.9092038513303655
News & Magazines : 2.6296718086758464
Social : 2.474376229423336
Books & Reference : 2.298374572937157
Travel & Local : 2.2569624184698207
Shopping : 2.0913138006004766
Simulation : 1.9981364530489698
Arcade : 1.9049591054974633
Dating : 1.760016564861787
Casual : 1.7082513717776169
Video Players & Editors : 1.6771922559271144
Maps & Navigation : 1.3562480588052592
Puzzle : 1.232011595403251
Food & Drink : 1.1595403250854126
Role Playing : 1.0870690547675743
Strategy : 0.9835386685992339
Racing : 0.9421265141318977
Auto & Vehicles : 0.8800082824308935
Libra

Top apps for iOS

In [28]:
display_table(apple[1:],11)

Games : 53.66124774211477
Entertainment : 7.433652910935113
Education : 6.294289287203002
Photo & Video : 4.849242740030569
Utilities : 3.4458802278727245
Health & Fitness : 2.501042100875365
Productivity : 2.473252744198972
Social Networking : 2.3204112824788106
Lifestyle : 2.0008336807002918
Music : 1.9174656106711132
Shopping : 1.6951507572599693
Sports : 1.5839933305543976
Book : 1.5562039738780047
Finance : 1.445046547172433
Travel : 1.1254689453939142
News : 1.0421008753647354
Weather : 1.0004168403501459
Reference : 0.8892594136445742
Food & Drink : 0.8753647353063776
Business : 0.7919966652771988
Navigation : 0.6391552035570377
Medical : 0.31957760177851885
Catalogs : 0.1389467833819647


As we can see, nobody from android can take top place - it's a mix between tools, games, entertainment apps, etc.

At iOS level however, Games is the most popular genre by far

We can see the avg number of ratings an app of a certain
genre got , for iOS

In [29]:
app_genres_ios = freq_table(apple[1:],11)
for genre in app_genres_ios:
    total = 0
    len_genre = 0
    for row in apple[1:]:
        genre_app = row[11]
        if genre_app == genre:
            total += float(row[5])
            len_genre +=1
    avg_no_ratings = total / len_genre
    print(str(genre) + ": " + str(avg_no_ratings))


Social Networking: 45498.89820359281
Photo & Video: 14352.280802292264
Games: 13691.996633868463
Music: 28842.021739130436
Reference: 22410.84375
Health & Fitness: 9913.172222222222
Weather: 22181.027777777777
Utilities: 6863.822580645161
Travel: 14129.444444444445
Shopping: 18615.32786885246
News: 13015.066666666668
Navigation: 11853.95652173913
Lifestyle: 6161.763888888889
Entertainment: 7533.678504672897
Food & Drink: 13938.619047619048
Sports: 14026.929824561403
Book: 5125.4375
Finance: 11047.653846153846
Education: 2239.2295805739514
Productivity: 8051.3258426966295
Business: 4788.087719298245
Catalogs: 1732.5
Medical: 592.7826086956521


we will now see how many installs each genre has for android

In [33]:
categories_android = freq_table(android_clean, 1)
for category in categories_android:
    total = 0
    len_category = 0
    for app in android_clean:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1856362.2950819673
AUTO_AND_VEHICLES : 625061.305882353
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 7504367.459459459
BUSINESS : 1659916.3452380951
COMICS : 803234.8214285715
COMMUNICATION : 35042146.82857143
DATING : 828971.2176470588
EDUCATION : 1775252.3364485982
ENTERTAINMENT : 11375402.298850575
EVENTS : 249580.640625
FINANCE : 1319851.4028985507
FOOD_AND_DRINK : 1891060.2767857143
HEALTH_AND_FITNESS : 3972300.388888889
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 630903.6904761905
LIFESTYLE : 1365375.4444444445
GAME : 14226135.745243128
FAMILY : 3319926.0965848453
MEDICAL : 96944.49873417722
SOCIAL : 22961790.384937238
SHOPPING : 6932419.727722772
PHOTOGRAPHY : 16577037.918149466
SPORTS : 3373767.6861538463
TRAVEL_AND_LOCAL : 13218662.767123288
TOOLS : 9774151.887816647
PERSONALIZATION : 4075783.994680851
PRODUCTIVITY : 15489549.114973262
PARENTING : 525351.8333333334
WEATHER : 4570892.658227848
VIDEO_PLAYERS : 23975016.585365854
NEWS_AND_MAGAZI

### Conclusion

The goal of this project was to familiarize ourself with pure python analysis, without Pandas or NumPy, and to find a suitable app genre from the datasets after cleaning them up. We can conclude that a game would probably be the best choice to try and enter the market with.